pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -0,0 +1,1097 @@
+            steps[2] = IBatchRouter.SwapPathStep({ pool: address(ybToken2), tokenOut: _token2Fork, isBuffer: true });
+        }
+    }
+
+    function _unbalanceBuffers(uint256 unbalancedToken1, uint256 unbalancedToken2) private {
+        if (unbalancedToken1 > _token1BufferInitAmount / 2) {
+            _unbalanceBuffer(WrappingDirection.WRAP, ybToken1, unbalancedToken1 - _token1BufferInitAmount / 2);
+        } else {
+            _unbalanceBuffer(WrappingDirection.UNWRAP, ybToken1, _token1BufferInitAmount / 2 - unbalancedToken1);
+        }
+
+        if (unbalancedToken2 > _token2BufferInitAmount / 2) {
+            _unbalanceBuffer(WrappingDirection.WRAP, ybToken2, unbalancedToken2 - _token2BufferInitAmount / 2);
+        } else {
+            _unbalanceBuffer(WrappingDirection.UNWRAP, ybToken2, _token2BufferInitAmount / 2 - unbalancedToken2);
+        }
+    }
+
+    function _unbalanceBuffer(WrappingDirection direction, IERC4626 wToken, uint256 amountToUnbalance) private {
+        if (amountToUnbalance < MIN_TRADE_AMOUNT) {
+            // If amountToUnbalance is very low, returns without unbalancing the buffer.
+            return;
+        }
+
+        IERC20 tokenIn;
+        IERC20 tokenOut;
+        uint256 exactAmountIn;
+
+        if (direction == WrappingDirection.WRAP) {
+            tokenIn = IERC20(wToken.asset());
+            tokenOut = IERC20(address(wToken));
+            exactAmountIn = amountToUnbalance;
+        } else {
+            tokenIn = IERC20(address(wToken));
+            tokenOut = IERC20(wToken.asset());
+            exactAmountIn = wToken.convertToShares(amountToUnbalance);
+        }
+
+        IBatchRouter.SwapPathStep[] memory steps = new IBatchRouter.SwapPathStep[](1);
+
+        steps[0] = IBatchRouter.SwapPathStep({ pool: address(wToken), tokenOut: tokenOut, isBuffer: true });
+
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = new IBatchRouter.SwapPathExactAmountIn[](1);
+        paths[0] = IBatchRouter.SwapPathExactAmountIn({
+            tokenIn: tokenIn,
+            steps: steps,
+            exactAmountIn: exactAmountIn,
+            minAmountOut: 0
+        });
+
+        vm.prank(lp);
+        batchRouter.swapExactIn(paths, MAX_UINT256, false, bytes(""));
+    }
+
+    function _setupTokens() private {
+        // Label deployed wrapped tokens.
+        vm.label(address(ybToken1), "ybToken1");
+        vm.label(address(ybToken2), "ybToken2");
+
+        // Identify and label underlying tokens.
+        _token1Fork = IERC20(ybToken1.asset());
+        vm.label(address(_token1Fork), "token1");
+        _token2Fork = IERC20(ybToken2.asset());
+        vm.label(address(_token2Fork), "token2");
+    }
+
+    function _setupLPAndVault() private {
+        vm.startPrank(donorToken1);
+        // Donate token1 (underlying) to LP.
+        _token1Fork.transfer(lp, 100 * _token1BufferInitAmount);
+        // Donate to vault, so it has enough tokens to wrap and do not preview.
```

Comment:
Can we document somewhere that this is not the worst-case scenario?

Reply:
I'll remove this, it's not needed anymore.